climt/_components/emanuel/component.py

```diff
@@ -242,40 +201,29 @@ def __init__(self,
         self._mf_timescale = reference_mass_flux_timescale
         self._ntracers = number_of_tracers
 
-        if specific_heat_condensible is not None:
-            specific_heat_condensible = DataArray(
-                specific_heat_condensible, attrs={'units': 'J kg^-1 K^-1'})
-
-        self._g = replace_none_with_default(
-            'gravitational_acceleration', acceleration_gravity)
+        self._g = get_constant(
+            'gravitational_acceleration')
 
-        self._Cpd = replace_none_with_default(
-            'heat_capacity_of_dry_air_at_constant_pressure',
-            specific_heat_dry_air)
+        self._Cpd = get_constant(
+            'heat_capacity_of_dry_air_at_constant_pressure')
 
-        self._Cpv = replace_none_with_default(
-            'heat_capacity_of_water_vapor_at_constant_pressure',
-            specific_heat_condensible)
+        self._Cpv = get_constant(
+            'heat_capacity_of_vapor_phase')
 
-        self._Rair = replace_none_with_default(
-            'gas_constant_of_dry_air',
-            gas_constant_dry_air)
+        self._Rair = get_constant(
```

Comment:
I'd encourage you to use Rdair or Rd to imply that it's the gas constant for dry air and not for air containing both dry and moist components.